airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io.airbyte.integrations.source.mongodb/MongoDbSource.java

```diff
@@ -109,20 +115,22 @@ protected List<TableInfo<CommonField<BsonType>>> discoverInternal(final MongoDat
     final List<TableInfo<CommonField<BsonType>>> tableInfos = new ArrayList<>();
 
     final Set<String> authorizedCollections = getAuthorizedCollections(database);
-    authorizedCollections.parallelStream().forEach(collectionName -> {
-      final MongoCollection<Document> collection = database.getCollection(collectionName);
-      final List<CommonField<BsonType>> fields = MongoUtils.getUniqueFields(collection).stream().map(MongoUtils::nodeToCommonField).toList();
-
-      // The field name _id is reserved for use as a primary key;
-      final TableInfo<CommonField<BsonType>> tableInfo = TableInfo.<CommonField<BsonType>>builder()
-          .nameSpace(database.getName())
-          .name(collectionName)
-          .fields(fields)
-          .primaryKeys(List.of(MongoUtils.PRIMARY_KEY))
-          .build();
-
-      tableInfos.add(tableInfo);
-    });
+    authorizedCollections.parallelStream()
+        .filter(this::isSupportedCollection)
```

Comment:
I wish that java had a [filterNot](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/filter-not.html)